pkg/runtime/depot.go

```diff
@@ -422,3 +465,39 @@ func someFilesExist(filePaths []string, basePath string) bool {
 	}
 	return false
 }
+
+// removeStaleArtifacts iterates over all unused artifacts in the depot, sorts them by last access
+// time, and removes them until the size of cached artifacts is under the limit.
+func (d *depot) removeStaleArtifacts() {
+	type artifact struct {
+		id   strfmt.UUID
+		info *artifactInfo
+	}
+	var totalSize int64
+	unusedArtifacts := make([]*artifact, 0)
+
+	for id, info := range d.config.Cache {
+		if !info.InUse {
+			totalSize += info.Size
+			unusedArtifacts = append(unusedArtifacts, &artifact{id: id, info: info})
+		}
+	}
+	logging.Debug("There are %d unused artifacts totaling %.1f MB in size", len(unusedArtifacts), float64(totalSize)/float64(MB))
+
+	sort.Slice(unusedArtifacts, func(i, j int) bool {
+		return unusedArtifacts[i].info.LastAccessTime < unusedArtifacts[j].info.LastAccessTime
+	})
+
+	for _, artifact := range unusedArtifacts {
+		if totalSize <= d.cacheSize {
+			break // done
+		}
+		logging.Debug("Removing cached artifact '%s', last accessed on %s", artifact.id.String(), time.Unix(artifact.info.LastAccessTime, 0).Format(time.UnixDate))
```

Comment:
The debug line above seems useful to have in the log going forward, but this one would probably be a bit spammy.